leshan-integration-tests/credentials/generate_credentials.sh

```diff
@@ -104,7 +104,7 @@ keytool -genkeypair -alias server -keyalg EC -dname 'CN=localhost' \
         -storetype $DEFAULT_STORE_TYPE \
         -ext BasicConstraints=ca:false \
         -ext KeyUsage:critical=digitalSignature,keyAgreement \
-        -ext ExtendedkeyUsage=serverAuth \
+        -ext ExtendedkeyUsage=serverAuth,clientAuth \
```

Comment:
Just a note -- making multi purpose certificates can hide problems from real deployments where one should not have multi purpose certificates. -- in server it is easier but especially in client you do not have necessary information to verify that device is acting as a server. That is the reason why I made them explicit.

Reply:
As explained [here](https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/410), I can not see how we can make "none queue mode" work in reliable way without supporting both role at both side. 

So in my mind, "multi purpose" is the default behavior.

But I can imagine that I maybe missed some real life constraint about x509 world. 

So could you elaborate about : 

> "in server it is easier but especially in client you do not have necessary information to verify that device is acting as a server" ? 


About : 
> making multi purpose certificates can hide problems from real deployments where one should not have multi purpose certificates.

A solution could be to add some test with clientOnly  ?

Reply:
@dachaac, we can continue the discussion at #997 if you want.